config/zfs-build.m4

```diff
@@ -205,6 +205,21 @@ AC_DEFUN([ZFS_AC_DEBUG_INVARIANTS], [
 	AC_MSG_RESULT([$enable_invariants])
 ])
 
+dnl # Disabled by default. If enabled allows a configured "turn objtools
+dnl # warnings into errors" (CONFIG_OBJTOOL_WERROR) behavior to take effect.
+dnl # If disabled, objtool warnings are never turned into errors.
+dnl #
+AC_DEFUN([ZFS_AC_OBJTOOL_WERROR], [
+	AC_MSG_CHECKING([whether objtool error on warning behavior is enabled])
+	AC_ARG_ENABLE([objtool-werror],
+		[AS_HELP_STRING([--enable-objtool-werror],
+		[Enable objtool's error on warning behaviour if present @<:@default=no@:>@])],
+		[enable_objtool_werror=yes],
+		[])
+
+	AC_MSG_RESULT([$enable_objtool_werror])
```

Comment:
When it's disabled we should output `no` in the `configure` output.  Probably the easiest thing to do here is set it to `no` and handle the case elsewhere.

```
checking whether FreeBSD kernel INVARIANTS checks are enabled... no
checking whether objtool error on warning behavior is enabled...    <<< should be yes or no
checking that generated files are newer than configure... done
```

It'd be nice to make `./configure --enable-objtool-werror` fatal when the `objtool` doesn't support `--Werror`.  It seems like we probably want this check after all.

Reply:
Right, will fix.

Reply:
I had the impression that there's room for misunderstanding here. Wouldn't it be even better to fail if `CONFIG_OBJTOOL_WERROR` isn't set? That would ensure that people expecting this to turn on -Werror on any system will notice?

Reply:
Right, what we want is for `--enable-objtool-werror` to be fatal if it's requested but can't be used.  As you said, that way folks will immediately notice.  Checking for `CONFIG_OBJTOOL_WERROR` would cover it, unless we also want to support adding the `--Werror` flag ourselves even when it's not enabled for the rest of the kernel.  It seems like this would work as long as `objtool` supports it.

Reply:
It would only make sense if we'd be objtool clean though. Maybe something to consider once that happens.